packages/gen-ai-hub/src/orchestration/orchestration-client.ts

```diff
@@ -16,25 +13,31 @@ export class OrchestrationClient {
   /**
    * Creates a completion for the chat messages.
    * @param data - The input parameters for the chat completion.
-   * @param deploymentResolver - A deployment ID or a function to retrieve it.
+   * @param deploymentId - A deployment ID or undefined to retrieve it based on the given model.
    * @param requestConfig - Request configuration.
    * @returns The completion result.
    */
   async chatCompletion(
     data: OrchestrationCompletionParameters,
-    deploymentResolver: DeploymentResolver = () =>
-      resolveDeployment({ scenarioId: 'orchestration' }),
+    deploymentId?: string,
     requestConfig?: CustomRequestConfig
   ): Promise<CompletionPostResponse> {
     const body = constructCompletionPostRequest(data);
-    const deployment =
-      typeof deploymentResolver === 'function'
-        ? (await deploymentResolver()).id
-        : deploymentResolver;
+    deploymentId =
+      deploymentId ??
+      (
+        await resolveDeployment({
+          scenarioId: 'orchestration',
```

Comment:
[q] In this case, we don't need to consider the resource group passed in request config (if)?